src/client/options/mod.rs

```diff
@@ -846,22 +846,40 @@ impl TlsOptions {
             };
 
             file.seek(SeekFrom::Start(0))?;
```

Comment:
since we're only taking the first one, it may be a bit clearer to perform this whole process in a single loop which can exit early using `read_one`. The `rustls` [example code](https://github.com/rustls/rustls/blob/main/rustls-mio/examples/tlsclient.rs#L420) actually has a good snippet for this:
```rust
fn load_private_key(filename: &str) -> rustls::PrivateKey {
    let keyfile = fs::File::open(filename).expect("cannot open private key file");
    let mut reader = BufReader::new(keyfile);

    loop {
        match rustls_pemfile::read_one(&mut reader).expect("cannot parse private key .pem file") {
            Some(rustls_pemfile::Item::RSAKey(key)) => return rustls::PrivateKey(key),
            Some(rustls_pemfile::Item::PKCS8Key(key)) => return rustls::PrivateKey(key),
            None => break,
            _ => {}
        }
    }

    panic!(
        "no keys found in {:?} (encrypted keys not supported)",
        filename
    );
}
```
I looks like we may need to add the `rustls-pemfile` crate as a dependency for this though, since it seems that `rustls` broke out the logic for reading these files into a separate crate. It's likely we have it as a transitive dependency already.

Reply:
Updated.